xtuner/dataset/refcoco.py

```diff
@@ -0,0 +1,493 @@
+    conversation['system'] = ""
+    return {'conversation': [conversation]}
+
+
+class RefCOCOTrainDataset(Dataset):
+    def __init__(self,
+                 data_path,  # path to refcoco path
+                 image_folder,
+                 tokenizer,
+                 processor,
+                 dataset='refcoco',
+                 splitBy='unc',
+                 image_size=448,
+                 max_dataset_length=None,
+                 dataset_map_fn=None,
+                 template_map_fn=None,
+                 max_length=2048,
+                 image_aspect_ratio='pad'):
+        self.vis_root = image_folder
+        self.text_processor = BlipCaptionProcessor(max_words=50)
+
+        if isinstance(processor, dict) or isinstance(
+                processor, Config) or isinstance(processor, ConfigDict):
+            self.processor = BUILDER.build(processor)
+        else:
+            self.processor = processor
+
+        self.refer = REFER(data_path, image_folder, dataset, splitBy)
+        self.ref_ids = self.refer.getRefIds(split="train")
+
+        self.instruction_pool = [
+            "[refer] {}",
+            "[refer] give me the location of {}",
+            "[refer] where is {} ?",
+            "[refer] from this image, tell me the location of {}",
+            "[refer] the location of {} is",
+            "[refer] could you tell me the location for {} ?",
+            "[refer] where can I locate the {} ?",
+        ]
+
+        def refcoco_prepare_hf(data):
+            json_data = DatasetDict({'train': HFDataset.from_list([data])})
+
+            data_set = process_hf_dataset(dataset=json_data,
+                                          tokenizer=tokenizer,
+                                          max_length=max_length,
+                                          dataset_map_fn=dataset_map_fn,
+                                          template_map_fn=template_map_fn,
+                                          split='train',
+                                          max_dataset_length=max_dataset_length,
+                                          remove_unused_columns=False,
+                                          pack_to_max_length=False,
+                                          with_image_token=True
+                                          )
+            return data_set[0]
+
+        self.prepare_hf_datasets = refcoco_prepare_hf
+
+    def preprocess(self, index):
+        ref_id = self.ref_ids[index]
+        ref = self.refer.loadRefs(ref_id)[0]
+
+        image_file = 'COCO_train2014_{:0>12}.jpg'.format(ref["image_id"])
+        image_path = os.path.join(self.vis_root, image_file)
+        image = Image.open(image_path).convert("RGB")
+        image_orig_size = image.size
+        image = self.processor.preprocess(
+            image, return_tensors='pt')['pixel_values'][0]
+        image_new_size = [image.shape[1], image.shape[2]]
+
+        image_new_size = [100, 100]
```

Comment:
Why modify the image_new_size to [100, 100] here? To get the percentage coordinates?

Reply:
It normalizes the coordinates to [0,100]. This is the default setting of minigpt4. There may be a unified argument to set coordinate normalization methods after we add several coordinate-related datasets.